rust/noosphere/src/ffi/error.rs

```diff
@@ -1,13 +1,86 @@
-use safer_ffi::prelude::*;
-
 use crate::error::NoosphereError;
+use safer_ffi::prelude::*;
 
 impl From<NoosphereError> for repr_c::Box<NsError> {
     fn from(error: NoosphereError) -> Self {
         Box::new(NsError { inner: error }).into()
     }
 }
 
+const NOOSPHERE_ERROR_OTHER: u8 = 1;
+const NOOSPHERE_ERROR_NETWORK_OFFLINE: u8 = 2;
+const NOOSPHERE_ERROR_NO_CREDENTIALS: u8 = 3;
+const NOOSPHERE_ERROR_MISSING_CONFIGURATION: u8 = 4;
+
+#[ffi_export]
+#[derive_ReprC(rename = "ns_error_code")]
+#[repr(u8)]
+/// Constant values for error codes from ns_error_t.
+pub enum NsErrorCode {
```

Comment:
Given that the representation on the Swift side *must* be `u32`, let's change all the representations to reflect that. Otherwise it's a bit confusing.